src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/EmbeddedLanguages/VirtualChars/AbstractVirtualCharService.cs

```diff
@@ -125,13 +126,13 @@ private void CheckInvariants(SyntaxToken token, VirtualCharSequence result)
 
             var tokenText = token.Text;
 
-            if (startDelimiter.Length > 0 && !tokenText.StartsWith(startDelimiter))
+            if (startDelimiter.Length > 0 && !tokenText.StartsWith(startDelimiter, StringComparison.Ordinal))
```

Comment:
assert fired before in cases with interesting unicode strings where the local-based StartsWith was not the same as the Ordinal StartsWith.